src/algebra/quaternion.lean

```diff
@@ -573,3 +583,55 @@ monoid_with_zero_hom.map_div norm_sq a b
 end field
 
 end quaternion
+
+private theorem one_le_four : 1 ≤ 4 := by simp
+
+namespace cardinal
+
+open_locale cardinal quaternion
+
+section quaternion_algebra
+
+variables {R : Type*} (c₁ c₂ : R)
+
+private theorem pow_four [infinite R] : #R ^ 4 = #R :=
+let H : 1 ≤ 4 := by simp in power_nat_eq (omega_le_mk R) H
+
+/-- The cardinality of a quaternion algebra, as a type. -/
+@[simp] lemma mk_quaternion_algebra : #ℍ[R, c₁, c₂] = #R ^ 4 :=
+by { rw mk_congr (quaternion_algebra.equiv_prod c₁ c₂), simp only [mk_prod, lift_id], ring }
+
+@[simp] lemma mk_quaternion_algebra_of_infinite [infinite R] : #ℍ[R, c₁, c₂] = #R :=
```

Comment:
This way the simp set will not be confluent. So it might be better to only mark the second one as `simp`. Or to mark the `pow_four` lemma as simp (and make it non-private). But really, finite quaternion algebras are boring, because of Wedderburns little theorem https://en.wikipedia.org/wiki/Wedderburn%27s_little_theorem
They turn out to be either fields or matrix algebras. So I don't think anyone will really expect `simp` to calculate their cardinalities.